lib/postinstall.js

```diff
@@ -0,0 +1,115 @@
+var fs = require("fs");
+var path = require("path");
+var appRoot = "../../";
+
+/**
+ * Create dev tools gradle runtime entry
+ */
+function writeFabricServiceGradleHook() {
+    console.log("Install Onesignal-build-gradle hook.");
+    try {
+        if (!fs.existsSync(path.join(appRoot, "hooks"))) {
+            fs.mkdirSync(path.join(appRoot, "hooks"));
+        }
+        if (!fs.existsSync(path.join(appRoot, "hooks", "after-prepare"))) {
+            fs.mkdirSync(path.join(appRoot, "hooks", "after-prepare"));
+        }
+        var scriptContent =
+            `
+var path = require("path");
+var fs = require("fs");
+module.exports = function($logger, $projectData, hookArgs) {
+  var platform = hookArgs.platform.toLowerCase();
+
+  function updateAppGradleScript(file) {
+    var appBuildGradleContent = fs.readFileSync(file).toString();
+    if (!appBuildGradleContent.match(/.*onesignal.*/)) {
+      $logger.trace("Configuring Onesignal for Android");
+      var search = -1;
+
+      var repositories = appBuildGradleContent.match(/repositories \{([^}]+)\}/gm);
+      if (!repositories) {
+          return;
+      }
+      search = appBuildGradleContent.indexOf(repositories[0]) + repositories[0].length;
+      appBuildGradleContent = appBuildGradleContent.substr(0, search) + \`\n	      maven { url "https://plugins.gradle.org/m2/" }\n\` + appBuildGradleContent.substr(search);
+
+      search = appBuildGradleContent.indexOf("apply plugin: \\"com.android.application\\"");
+      if (search == -1) {
+          return;
+      }
+      appBuildGradleContent = appBuildGradleContent.substr(0, search + 39) + \`\napply plugin: "com.onesignal.androidsdk.onesignal-gradle-plugin"\n\` + appBuildGradleContent.substr(search + 39);
```

Comment:
Adding the `onesignal-gradle-plugin` to `platforms/android/include.gradle` worked just fine in testing with tns-android 4.1.1. As [PR 36](https://github.com/roblav96/nativescript-onesignal/pull/36/files#diff-46fdbd72a3c8c0bca1acb1d908920bfb) suggests

It is possible that older versions of tns-android may require that `apply plugin: "com.onesignal.androidsdk.onesignal-gradle-plugin"` to be before `apply plugin: "com.android.application"` so a solution like this string inserting might be required.